packages/grid/src/vaadin-grid-column-group.js

```diff
@@ -236,6 +237,42 @@ class GridColumnGroup extends ColumnBaseMixin(PolymerElement) {
     this._setFlexGrow(Array.prototype.reduce.call(this._visibleChildColumns, (prev, curr) => prev + curr.flexGrow, 0));
   }
 
+  /**
+   * This method is called before the group's frozen value is being propagated to the child columns.
+   * In case some of the child columns are frozen, while others are not, the non-frozen ones
+   * will get automatically frozen as well. As this may sometimes be unintended, this method
+   * shows a warning in the console in such cases.
+   * @private
+   **/
+  __scheduleAutoFreezeWarning(columns, frozenProp) {
+    if (this._grid) {
+      // Derive the attribute name from the property name
+      const frozenAttr = frozenProp.replace(/([A-Z])/g, '-$1').toLowerCase();
+
+      // Check if all the columns have the same frozen value
+      const firstColumnFrozen = columns[0][frozenProp] || columns[0].hasAttribute(frozenAttr);
+      const allSameFrozen = columns.every((column) => {
+        return (column[frozenProp] || column.hasAttribute(frozenAttr)) === firstColumnFrozen;
+      });
+
+      if (!allSameFrozen) {
+        // Some of the child columns are frozen, some are not. Show a warning.
+        this._grid.__autoFreezeWarningDebouncer = Debouncer.debounce(
+          this._grid.__autoFreezeWarningDebouncer,
+          animationFrame,
+          () => {
+            console.warn(
+              'WARNING: Joining frozen and non-frozen Grid columns inside the same column group! ' +
```

Comment:
Should we use `frozenProp` in the warning to make it clear which property causes a problem?

Reply:
Done